src/plugins/ncr/components/labels-picker/index.jsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import chroma from 'chroma-js';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import isEqual from 'lodash/isEqual';
+import labelColors from 'config/labels'; // eslint-disable-line import/no-unresolved
+import Message from '@gdbots/pbj/Message';
+import createDelegateFactory from '@triniti/app/createDelegateFactory';
+import {
+  Button,
+  Card,
+  CardBody,
+  Col,
+  Icon,
+  Row,
+} from '@triniti/admin-ui-plugin/components';
+import PicklistPicker from '@triniti/cms/plugins/sys/components/picklist-picker';
+import delegateFactory from './delegate';
+
+const getColor = (label) => labelColors[label] || '#c0c0c0';
+
+class Labels extends React.PureComponent {
+  static propTypes = {
+    disabled: PropTypes.bool,
+    disabledReasonMessage: PropTypes.string,
+    delegate: PropTypes.shape({
+      handleApplyLabels: PropTypes.func.isRequired,
+    }).isRequired,
+    node: PropTypes.instanceOf(Message),
+  };
+
+  static defaultProps = {
+    disabled: false,
+    disabledReasonMessage: 'Labels disabled until saved.',
+    node: null,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      touched: false,
+      selected: [],
+    };
+
+    this.isNodeDiff = this.isNodeDiff.bind(this);
+    this.handleApplyLabels = this.handleApplyLabels.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  UNSAFE_componentWillReceiveProps({ node }) {
```

Comment:
@rsumilang can probably use the non-deprecated `componentDidUpdate` here

Reply:
@efjacobson `componentDidUpdate` doesn't allow me to use setState.

Reply:
@rsumilang i think it does, you just have to be careful to not create an infinite render loop https://github.com/wb-apps/tmz-web/blob/master/client/components/gallery-widget/lightbox/index.jsx#L141